corehq/apps/app_manager/views.py

```diff
@@ -908,9 +909,19 @@ def get_details(case_type_):
         def _report_to_config(report):
             return {
                 'report_id': report._id,
-                'title': report.title
+                'title': report.title,
+                'charts': [chart for chart in report.configured_charts if chart['type'] == 'multibar'],
             }
         all_reports = ReportConfiguration.by_domain(app.domain)
+        for report in all_reports:
+            for i in range(len(report.configured_charts)):
+                chart_id = report.configured_charts[i].get('chart_id')
+                if not chart_id:
+                    report.configured_charts[i]['chart_id'] = (
+                        report.configured_charts[i].get('title') or ''
+                    ) + '-' + uuid.uuid4().hex
+            report.save()
```

Comment:
picturing something more like this: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/reports/specs.py#L101 which you can add to `ChartSpec` (and then work with wrapped objects in the template)